src/Maker/AdminMaker.php

```diff
@@ -0,0 +1,249 @@
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        $io->section('Welcome to the Sonata admin maker');
+        $modelClass = $io->ask(
+            'The fully qualified model class',
+            $input->getArgument('model'),
+            'Sonata\AdminBundle\Command\Validators::validateClass'
+        );
+        $modelClassBasename = current(array_slice(explode('\\', $modelClass), -1));
+
+        $adminClassBasename = $io->ask(
+            'The admin class basename',
+            $input->getOption('admin') ?: $modelClassBasename.'Admin',
+            'Sonata\AdminBundle\Command\Validators::validateAdminClassBasename'
+        );
+        if (\count($this->availableModelManagers) > 1) {
+            $managerTypes = array_keys($this->availableModelManagers);
+            $managerType = $io->choice('The manager type', $managerTypes, $managerTypes[0]);
+
+            $input->setOption('manager', $managerType);
+        }
+        if ($io->confirm('Do you want to generate a controller?', false)) {
+            $controllerClassBasename = $io->ask(
+                'The controller class basename',
+                $input->getOption('controller') ?: $modelClassBasename.'AdminController',
+                'Sonata\AdminBundle\Command\Validators::validateControllerClassBasename'
+            );
+            $input->setOption('controller', $controllerClassBasename);
+        }
+        if ($io->confirm('Do you want to update the services YAML configuration file?', true)) {
+            $path = $this->projectDirectory.'/config/';
+            $servicesFile = $io->ask(
+                'The services YAML configuration file',
+                is_file($path.'admin.yaml') ? 'admin.yaml' : 'services.yaml',
+                'Sonata\AdminBundle\Command\Validators::validateServicesFile'
+            );
+            $id = $io->ask(
+                'The admin service ID',
+                $this->getAdminServiceId($adminClassBasename),
+                'Sonata\AdminBundle\Command\Validators::validateServiceId'
+            );
+            $input->setOption('services', $servicesFile);
+            $input->setOption('id', $id);
+        } else {
+            $input->setOption('services', false);
+        }
+        $input->setArgument('model', $modelClass);
+        $input->setOption('admin', $adminClassBasename);
+    }
+
+    /**
+     * Configure any library dependencies that your maker requires.
+     *
+     * @param DependencyBuilder $dependencies
+     */
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        // TODO: Implement configureDependencies() method.
+    }
+
+    /**
+     * Called after normal code generation: allows you to do anything.
+     *
+     * @param InputInterface $input
+     * @param ConsoleStyle   $io
+     * @param Generator      $generator
+     */
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
```

Comment:
this is a very big method, can you maybe break it into smaller ones with name self-explanatory names.